index.js

```diff
@@ -518,6 +518,11 @@ module.exports = class Hypercore extends EventEmitter {
     return await this.core.append(buffers, this.sign, { preappend })
   }
 
+  async treeHash (length = this.length) {
+    const roots = await this.core.tree.getRoots(length)
```

Comment:
you should call `await this.ready()` internally before defaulting the length so it's loaded